components/addFileButton/index.jsx

```diff
@@ -43,8 +43,9 @@ export function handleSaveResource({
 }) {
   return async (uploadedFile) => {
     try {
-      const response = await overwriteFile(
-        joinPath(currentUri, encodeURIComponent(uploadedFile.name)),
+      const fileName = uploadedFile.name.replace(/^-/, "");
```

Comment:
Doing this here would prevent the "Do you want to overwrite the existing file?" dialog from popping up, if `foo` exists and you try to upload `-foo`.